docs: guide for Chrome DevTools workspace mapping with Vite-Plugin-Devtools-Json (#3103)
wasp-lang/wasp: web/docs/project/custom-vite-config.md

    ```
  </TabItem>
</Tabs>

> ## Chrome DevTools automatic workspace mapping {#devtools-workspace}

Reviewer: Should be 3 `###`, also changed a bit the name so it is more clear what it accomplishes

```suggestion
### Editing from the Chrome DevTools {#devtools-workspace}
```

> Want edits you make in Chrome DevTools Sources panel to save directly to your project files? You can enable [Chrome DevTools Workspace](https://developer.chrome.com/docs/devtools/workspace/) support in development with the community plugin **vite-plugin-devtools-json** (no Wasp fork required).

Reviewer: Rewrote it a bit. We tell people want they can accomplish, and what they'll have to do to accomplish it. This way docs are super nice to read! I also added a link to the plugin's page.

```suggestion
Chrome DevTools support [mapping a page's resources to a folder](https://developer.chrome.com/docs/devtools/workspaces), so any changes you make in the browser reflect back to your files. To enable it, you can add their Vite plugin: [`vite-plugin-devtools-json`](https://github.com/ChromeDevTools/vite-plugin-devtools-json).
```

1. Install the plugin as a **dev-dependency**:

```bash

> npm add -D vite-plugin-devtools-json

Reviewer: When using `npm`, this command is usually done with `i` or `install` instead of `add`. And everywhere else in our docs we use `i`, so let's change it.

```suggestion
npm i -D vite-plugin-devtools-json
```

```

2. Extend your `vite.config.{ts,js}`:

<Tabs groupId="js-ts">
  <TabItem value="js" label="JavaScript">

```js title="vite.config.js"
import { devtoolsJson } from 'vite-plugin-devtools-json'

export default {
  plugins: [
    react(),              // already present
    devtoolsJson({
      // tell Chrome which folder to map – project root works great
      root: __dirname
    })
  ]
}
```

  </TabItem>
  <TabItem value="ts" label="TypeScript">

```ts title="vite.config.ts"
import { defineConfig } from 'vite'
import react from '@vitejs/plugin-react-swc'
import { devtoolsJson } from 'vite-plugin-devtools-json'

export default defineConfig({
  plugins: [
    react(),
    devtoolsJson({ root: __dirname })
  ]
})
```
  </TabItem>

> </Tabs>

Reviewer: - We have a plugin that lets us write only the TS version and converts it to JS automatically. You just write a normal code block and add the `auto-js` attribute to it.

- This file is an ES Module, not a CommonJS file, so `__dirname` is not the recommended way to get the dirname. We should use [`import.meta.dirname`](https://nodejs.org/docs/latest-v22.x/api/esm.html#importmetadirname) instead.

- We don't require users to add the `react()` plugin, that should be taken out.

- `devtoolsJson` is a _default_ export

~~~suggestion
```ts title="vite.config.ts" auto-js
import { defineConfig } from 'vite'
import devtoolsJson from 'vite-plugin-devtools-json'

export default defineConfig({
  plugins: [
    devtoolsJson({ root: import.meta.dirname })
  ]
})
```
~~~

> 3. Start your app with `wasp start`, open Chrome DevTools → **Sources › Filesystems** and you should see your project automatically mapped. Changes you make in DevTools now save to disk and Vite’s HMR updates the browser instantly.

Reviewer: - Let's make formatting consistent
- The folders are in Sources -> Workspace

```suggestion
3. Start your app with `wasp start`, open **Chrome DevTools → Sources → Workspace** and you should see your project automatically mapped. Changes you make in DevTools now save to disk and Vite's HMR updates the browser instantly.
```

:::tip Path normalisation

> The latest version of `vite-plugin-devtools-json` includes Windows, WSL and Docker-desktop path fixes contributed by the Wasp community – make sure you are on ≥ 0.4.0.

Reviewer: Small nitpick

```suggestion
The latest version of `vite-plugin-devtools-json` includes Windows, WSL and Docker Desktop path fixes contributed by the Wasp community – make sure you are on version 0.4.0 or greater.
```
